kedro/config/common.py

```diff
@@ -103,7 +107,8 @@ def _get_config_from_patterns(
             Path(conf_path), patterns, processed_files, _config_logger
         )
         new_conf = _load_configs(
-            config_filepaths=config_filepaths, ac_template=ac_template
+            config_filepaths=config_filepaths,
+            anyconfig_args=anyconfig_args if anyconfig_args else DEFAULT_ANYCONFIG_ARGS,
```

Comment:
The right logic here would be
```
{**DEFAULT_ANYCONFIG_ARGS, **anyconfig_args}
```
Otherwise you need to always repeat the keys given in `DEFAULT_ANYCONFIG_ARGS` when specifying `anyconfig_args` even if you're not modifying them from their default values.